rfcs/20230801-max-partition-policy/README.md

```diff
@@ -0,0 +1,116 @@
+of optimizations to be applied when partitioning.
+Currently, oneDNN Graph API supports two partition policies:
+
+```c
+typedef enum {
+    dnnl_graph_partition_policy_fusion = 1,
+    dnnl_graph_partition_policy_debug = 2,
+} dnnl_graph_partition_policy_t;
+```
+
+```cpp
+class partition {
+    ...
+    enum class policy {
+        fusion = dnnl_graph_partition_policy_fusion,
+        debug = dnnl_graph_partition_policy_debug,
+    };
+    ...
+};
+```
+
+- Fusion policy (default partition policy): fusion policy returns partitions
+  with [pre-defined fusion patterns](https://oneapi-src.github.io/oneDNN/dev_guide_graph_fusion_patterns.html),
+  e.g. Convolution followed by a chain of post-ops, MHA, MLP, etc.
+- Debug policy: debug policy returns partitions with single operation in each partition.
+  This policy doesn't apply any fusions, which is useful when
+  user wants to debug or compare performance with fusion policy.
+
+## Proposal
+
+The API to expose max policy to users is similar to fusion and debug policies.
+
+```c
+typedef enum {
+    dnnl_graph_partition_policy_max = 0, // the new max policy
+    dnnl_graph_partition_policy_fusion = 1,
+    dnnl_graph_partition_policy_debug = 2,
+} dnnl_graph_partition_policy_t;
+```
+
+```cpp
+class partition {
+    ...
+    enum class policy {
+        max = dnnl_graph_partition_policy_max, // the new max policy
+        fusion = dnnl_graph_partition_policy_fusion,
+        debug = dnnl_graph_partition_policy_debug,
+    };
+    ...
+};
+```
+
+The expected behaviors when using max policy:
+
+- If not specified, the default partition policy remains to be fusion policy.
+- Max policy returns partitions max in partition size. The criteria of partitioning
+  is based on pre-defined ops list and pre-defined rules. Connected ops
+  that meet the conditions of pre-defined ops list and pre-defined rules
+  will be selected into one partition.
+- The pre-defined ops list and pre-defined rules are backend specific.
+
+## Open questions
+
+1. Compared with fusion and debug policy, max policy gives the library more control
+   on partitioning. This on the other hand means users may lose fine-grained control
+   of what optimizations they want to get from the library.
+1. The performance of max policy may or may not be better than fusion policy. The
+   library tries to provide the best performance by model performance heuristics.
+   But for untested models, it is possible that the performance of max policy is
+   worse than fusion policy.
+
+## Known issues
+
+1. DNNL backend will not support max partition policy in oneDNN v3.3 release.
+   So when running with a combination of max partition policy and DNNL backend,
+   max partition policy will fallback to fusion partition policy.
+1. The validation for max partition policy will not be landed in oneDNN v3.3
+   release, so the implementation of max partition policy in Graph Compiler
+   backend is a experimental feature.
```

Comment:
Though validation cannot be covered in v3.3, it's still needed to describe the validation methodology and potential options here. I see @chuanqi129 is suggesting to add more arguments to benchdnn command line. It would be better to include that into this RFC also.

Reply:
good suggestion, will add.